Add DatabaseVulnerabilityAssessments swagger

ggildin opened the pull request:
The put api is updated.
A new parameter is added to its body properties storageAccountAccessKey.
storageContainerSasKey isn't required anymore, but the user must to specified one of the following parameters:
storageContainerSasKey
or
storageAccountAccessKey

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

ggildin committed:
Add DatabaseVulnerabilityAssessments swagger

The put api is updated.
A new parameter is added to its body properties storageAccountAccessKey.
storageContainerSasKey isn't required anymore, but the user must to specified one of the following parameters:
storageContainerSasKey
or
storageAccountAccessKey

AutorestCI commented:
# Automation for azure-sdk-for-python

AutorestCI commented:
# Automation for azure-sdk-for-java

AutorestCI commented:
# Automation for azure-sdk-for-ruby

AutorestCI commented:
# Automation for azure-sdk-for-go

AutorestCI commented:
# Automation for azure-sdk-for-node

jaredmoo commented:
Please look at travis CI failures

jaredmoo commented:
Are api-versions other than 2017-10 affected?

ggildin committed:
Add two missing example files

ggildin commented:
Hi.
1. I took a look on CI failures, hope that are fixed now.
2. Since it only add an optional (exclusive or) parameter, we don't think that a new api version is required.

lmazuel commented:
@ggildin @jaredmoo The new file is didn't plug into any tags, is this on purpose?

jaredmoo commented:
Let me rephrase my question. Here you are adding api-version 2017-10, but does api-version 2017-03 or earlier also contain the change?

ggildin committed:
update databaseVulnerabilityAssessments for version 2017-03

add storageContainerPath to properties as optional parameters

ggildin commented:
Hi.
I updated the 2017-03 documentation as well.
We don't have documentation at earlier versions.

ggildin commented:
@yaakoviyun

yaakoviyun commented on specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/databaseVulnerabilityAssessments.json:
```diff
@@ -261,12 +271,44 @@
       "allOf": [
         {
-          "$ref": "../../../common/v1/types.json#/definitions/ProxyResource"
+          "$ref": "#/definitions/ProxyResource"
```
put
-          "$ref": "../../../common/v1/types.json#/definitions/ProxyResource"

yaakoviyun commented on specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/databaseVulnerabilityAssessments.json:
```diff
@@ -226,7 +228,15 @@
+          "description": "Specifies the identifier key of the auditing storage account. If 'StorageContainerSasKey' isn't specified, storageAccountAccessKey is required.",
           "type": "string",
           "x-ms-mutability": [
             "create",
```
+            "create",
+            "update"?

ggildin committed:
Revert "update databaseVulnerabilityAssessments for version 2017-03"

This reverts commit 7f90a833cdce20259526b7d4e4bdf614eb000f75.

ggildin committed:
Revert "Add DatabaseVulnerabilityAssessments swagger"

This reverts commit 9f34855928e5b0e9aa638145a56f5d6c7daacb40.

ggildin committed:
Revert "Add two missing example files"

This reverts commit 660b2f79246ed83e855407aba5292857d264c2c8.

ggildin committed:
Update databaseVulnerabilityAssessments 2017-03-01 swagger

The put api is updated.
A new parameter is added to its body properties storageAccountAccessKey.
storageContainerSasKey isn't required anymore, but the user must to specified one of the following parameters:
storageContainerSasKey
or
storageAccountAccessKey

ggildin commented:
Hi.
1. I synced with @yaakoviyun, we don't want to exposed at frontend a new version of the api. So I removed the swagger file for 2017-10-01 versions. (The auth generated tool generated them).
2. I updated the 2017-03-01 version files.
3. After this changes no swagger file is added, so no tag updating is required.
3.  @jaredmoo - Regards Travis error - The failure is due to breaking change. Required field has been changed to be optional. As we talked in the CR, this is only a minor change and is acceptable.
Do I have any way to exclude this failure?

ravbhatnagar commented:
Looks good.

jaredmoo commented:
Please check travis errors:
- https://travis-ci.org/Azure/azure-rest-api-specs/jobs/396018461#L692
- https://travis-ci.org/Azure/azure-rest-api-specs/jobs/396018480#L741

ggildin committed:
Clean non required params in version 10-2017

clean databaseVulnerabilityAssessmentScans

jaredmoo commented:
There is still model validation error: https://travis-ci.org/Azure/azure-rest-api-specs/jobs/397699547

ggildin committed:
fix error in execute scan example

ggildin commented:
fixed

lmazuel commented:
@jaredmoo That's a preview Swagger, signed-of by ARM, and changes look good me. Once you think it's done, let me know and I merge the Pr.
Thanks!
